treeherder/services/signatures_remover.py

```diff
@@ -0,0 +1,146 @@
+import logging
+from typing import List
+
+from treeherder.perf.models import PerformanceDatum, PerformanceSignature
+from treeherder.services.taskcluster import TaskclusterModel
+from treeherder.services.max_runtime import MaxRuntime
+
+logger = logging.getLogger(__name__)
+
+RECEIVER_TEAM_EMAIL = "perftest-alerts@mozilla.com"
+RECEIVER_EMAIL = "dhunt@mozilla.com"
+
+
+class PublicSignaturesRemover:
+    """
+    This class handles the removal of signatures which are (no longer)
+    associated with a job and sends email notifications to the entire team.
+    """
+
+    table_description = """__Summary of recently deleted Performance Signatures__
+> The signatures below were deleted due to the fact that they were no longer associated with a job.
+---
+    """
+
+    table_headers = """
+| Repository | Framework | Platform | Suite | Application |
+| :---: | :---: | :---: | :---: | :---: |
+    """
+
+    def __init__(
+        self,
+        timer: MaxRuntime,
+        taskcluster_model: TaskclusterModel,
+    ):
+        self.tc_model = taskcluster_model
+        self._subject = "Summary of deleted Performance Signatures"
+        self._content = None
+        self._nr_of_rows_allowed = 50
+        self._nr_of_emails_allowed = 10
+
+        self.timer = timer
+
+    def remove_in_chunks(self, max_timestamp):
+        emails_sent = 0
+        rows_left = self._nr_of_rows_allowed
+        chunk_of_signatures = []
+
+        for performance_signature in PerformanceSignature.objects.filter(
+            last_updated__lte=max_timestamp
+        ):
+            self.timer.quit_on_timeout()
+
+            if emails_sent != self._nr_of_emails_allowed and (
```

Comment:
better to have `emails_sent <= self._nr_of_emails_allowed`

Reply:
Actually a simpler logic would be:
```
while emails_sent < self._nr_of_emails_allowed and (PerformanceDatum.objects.filter(...).exists()):
```
and you won't need `if emails_sent == self._nr_of_emails_allowed: break`

Reply:
If I look closely this is breaking the single responsibility principle: it is collecting the signature objects into chunks and then sending the email along with deleting the signatures collected. I would move the code under this `if` block into a function `collect_signatures` would return `chunk_of_signatures`.